api/src/main/java/io/druid/utils/CircularBuffer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.utils;
+
+import com.google.common.base.Preconditions;
+
+public class CircularBuffer<E>
+{
+  public E[] getBuffer()
+  {
+    return buffer;
+  }
+
+  private final E[] buffer;
+
+  private int start = 0;
+  private int size = 0;
+
+  public CircularBuffer(int capacity)
+  {
+    buffer = (E[]) new Object[capacity];
```

Comment:
Maybe explode with a precondition check that capacity is larger than 0 here instead of exploding out of bounds [here](https://github.com/druid-io/druid/pull/5418/files#diff-876531212c5ea38eb8111a49277bf9d9R43)

Reply:
Added a preconditions check